linkerd/buffer/src/dispatch.rs

```diff
@@ -201,20 +179,11 @@ mod test {
             assert!(dispatch.poll().unwrap().is_not_ready());
             assert!(ready_rx.get_ref().as_ref().unwrap().is_ready());
             Delay::new(Instant::now() + max_idle).map(move |()| {
-                assert!(dispatch.poll().unwrap().is_not_ready(),);
-                assert!(
-                    dispatch.inner.is_none(),
-                    "Did not drop inner service after timeout."
-                );
+                assert!(dispatch.poll().unwrap().is_ready());
                 assert!(
                     ready_rx.get_ref().is_err(),
                     "Did not advertise an error to consumers."
                 );
-                drop(ready_rx);
```

Comment:
Should we still include a test that asserts dropping all senders results in the `Dispatch` being ready?